packages/aragon-wrapper/src/cache/index.test.js

```diff
@@ -95,18 +95,56 @@ test('should observe the key\'s value for changes in the correct order', async (
   )
   // act
   const observable = instance.observe('counter', 1)
+
+  // make sure getItem is finished before emitting
+  await new Promise(resolve => setTimeout(resolve, 700))
+
   // assert
   let emissionNumber = 0
   observable.subscribe(value => {
     emissionNumber++
-    // first value should be 1 (the default) because getItem returns falsy
+    // first value should be 4 because new sets happen immediately
```

Comment:
> Do new sets happen immediately because when running the tests it uses in memory ([no indexedDB and localStorage](https://github.com/aragon/aragon.js/blob/cache-observe-race/packages/aragon-wrapper/src/cache/index.js#L23))?

Yeah, the tests are abstracting over the db and don't actually set anything in them; they just test the behaviour of the interfaces.